[48기 김진희] Refactor : 멘토 리뷰 반영

jininii opened the pull request:
## 1. 본 PR이 우리 팀의 웹 서비스 제품성에 어떠한 기여를 하였고, 사용자에게 어떠한 기대효과를 전달하는지 작성해주세요.
- 내 PR이 제품 내 어떠한 기능적인 배경/전후맥락 가운데 개발되었나요?

  (예시) 고객의 유입과 전환을 책임지는 회원 기능에서, 유저의 이탈을 방지하기 위해 input 창 바로 아래에서 실시간으로 적색의 경고 메세지를 전달 받습니다.

- 내 PR이 Merge 됨으로써 유저에게 전달되는 편익/기대효과는 무엇일까요?

  (예시) 유저는 실시간으로 스스로 오기입한 내용을 확인할 수 있기 때문에 잘못 계정 정보를 작성함으로써 발생될 수 있는 변수를 최소화할 수 있습니다. 이는 곧 유저 전환의 허들을 낮추어 줄 수 있습니다.



<br />

## 2. 이 브랜치에서 어떤 내용을 개발했는지 큰 제목과 그리고 상세 내역을 적어주세요.

(예시) 로그인 기능 추가

- (예시) email입력 `@`필요,password에 숫자로 8자이상 유효성 검사 추가
- (예시) 유효성 검사 통과시 버튼 컬러 변경 및 버튼 활성화

<br />

## 3. 개발한 화면을 캡쳐해서 첨부 해 주세요. (drag & drop 또는 첨부파일 추가)
<br />

-

<br />

## 4. 이 브랜치에서 개발하면서 느꼇던 개발 성장포인트를 적어주세요.

- (예시) 새로운 배열메서드 filter를 쓸 수 있게됐습니다.
- (예시) 상수데이터와 map 메서드로 반복되는 UI를 그릴 수 있게 됐습니다.
- (예시) fetch함수를 어떻게 쓰는지 알게됐습니다.
- (예시) 계산된 속성명 (computed property)을 쓰긴했는데 어떻게 돌아가는지 이해가 잘 안되서
  블로그에 정리해봐야겠습니다.
- (예시) 화살표 함수(arrow-function) 어떤 코드를 언제 생략할 수있는지 완벽 이해했습니다.
- (예시) 구조분해할당(Destructuring assignment)을 써보니 가독성도, 유지보수도 편할것같습니다.
  <br />

jininii committed:
Modify : 파일 이동 및 경로 수정

jininii committed:
Add : Mission 1 - 사용자 입력 데이터 저장 기능 구현

jininii committed:
Add : Mission 1 - 사용자 입력 데이터 저장 기능 구현_2

jininii committed:
Add : Mission 2 - 로그인 버튼 활성화 기능 구현

jininii committed:
Add : Mission 3 - 댓글 기능 구현

5yoonl commented on src/pages/jinheeKim/Login/Login.js:
```diff
@@ -0,0 +1,83 @@
+    userPw:"패스워드",
+  });
+
+  const handleInput = (event) => {
+    const {value, id} = event.target;
+    setUserInfo({...userInfo, [id]:value});
+  };
```
👍 좋습니다! 계속해서 이렇게 세션내용 바로바로 적용해주면서 본인의 지식으로 가져가주시면 좋겠습니다!

5yoonl commented on src/pages/jinheeKim/Login/Login.js:
```diff
@@ -0,0 +1,83 @@
+    setPw(event.target.value);
+  };
+  위 두 줄을 아래처럼 하나로 묶음
+  */
```
코드에 주석이 많습니다!
추후에 다시 학습할 용도로 남겨놓은 주석이라면, 해당내용은 따로 블로그나 개인 메모에 작성을 해 주시고, 코드를 github을 통해 올려주실때는 불필요한 코드를 최대한 줄여주세요!

5yoonl commented on src/pages/jinheeKim/Login/Login.js:
```diff
@@ -0,0 +1,83 @@
+    <div className="login">
+      <div className="box">
+        <h1>Westagram</h1>
+        <div className="input-txt">
```
```suggestion
        <div className="inputTxt">
```
위 코드 처럼 js의 기본 컨벤션은 `camelCase`입니다! className을 작성할 때에도 신경써주셔야합니다!

5yoonl commented on src/pages/jinheeKim/Login/Login.js:
```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import { useNavigate } from 'react-router-dom';
+import "../../../style/reset.scss";
+import "../../../style/common.scss";
+import "../../../style/variables.scss"
```
해당 scss파일들은 Index.js에서 import 해와서 전역으로 적용시키는게 좋습니다.

5yoonl commented on src/pages/jinheeKim/Login/Login.scss:
```diff
@@ -0,0 +1,96 @@
+@import '../../../style/variables.scss';
+@import '../../../style/reset.scss';
+@font-face {
+	font-family: "Lobster";
+	src: url('../assets/Lobster-Regular.ttf');
+}
```
마찬가지 입니다! font같은 속성들도 common.scss에 작성해서 전역으로 관리할 수 있겠습니다!

5yoonl commented on src/pages/jinheeKim/Login/Login.scss:
```diff
@@ -0,0 +1,96 @@
+	src: url('../assets/Lobster-Regular.ttf');
+}
+
+$point-color-blue: rgba(47, 170, 249, 1);
+$grey-color-193: rgba(193, 193, 193, 1);
+$grey-color-border: rgba(219, 219, 219, 1);
+$grey-color-bg: rgba(250, 250, 250, 1);
```
variables.scss파일에 있으면 될 변수들이네요!

5yoonl commented on src/pages/jinheeKim/Main/Main.js:
```diff
@@ -0,0 +1,226 @@
+import React, {useState} from "react";
+import "../../../style/reset.scss";
+import "../../../style/common.scss";
```
index.js에서 전역으로 import한다면 이렇게 파일마다 매번 import 하지않아도 됩니다!

5yoonl commented on src/pages/jinheeKim/Main/Main.js:
```diff
@@ -0,0 +1,226 @@
+import TextareaAutosize from 'react-textarea-autosize';
+
+const Main = () => {
+  const [comment, setComment] = useState('');
```
`comment`라는 변수명이 작성되어있는 comment들의 모음인지, 내가 input에 입력한 값을 관리하는 변수인지 이름만 봤을 때 바로 추측하기 어렵습니다.

아래에 commentArray라는 변수도 마찬가지로 조금 더 변수가 어떤 state를 가지고 있는지 명확한 이름으로 작성해주시면 좋겠습니다

5yoonl commented on src/pages/jinheeKim/Main/Main.js:
```diff
@@ -0,0 +1,226 @@
+    if (comment === '') {
+      return;
+    }
+    setCommentArray(commentList => [comment, ...commentList]);
```
```suggestion
    setCommentArray(commentList => [ ...commentList,comment]);
```
이 두개의 차이점을 확실히 알고 넘어가주셨으면 좋겠습니다!!

5yoonl commented on src/pages/jinheeKim/Main/Main.js:
```diff
@@ -0,0 +1,226 @@
+  const onSubmit = event => {
+    event.preventDefault();
+    if (comment === '') {
+      return;
```
Input에 값이 비어있을때 함수가 그냥 아무런 동작을 하지 않는다면 기능상 에러인지, 의도된 동작인지 파악하기 어렵습니다.
`comment` 값이 비어있다면 함수가 호출될 때 `alert`를 표시해주는 등의 추가적인 조치가 있으면 좋습니다

5yoonl commented on src/pages/jinheeKim/Main/Main.js:
```diff
@@ -0,0 +1,226 @@
+                </div>
+                <form className="comment" onSubmit={onSubmit}>
+                    {commentArray.map((value, user) => (
+                        <p key={user} className="comment-list">
```
map 메서드에 콜백함수의 두번 째 인자, 즉 진희님의 코드에서 user라고하는 변수는 기준배열(commnetArray)의 value(배열의 각각의 요소)마다의 index(배열내의 순서)를 나타냅니다.  리액트에서 반복문을 사용해서 key 속성을 꼭 부여하라고 하는 이유는 해당요소를 리액트가 명확이 인식하기 위함인데, 배열의 index를 key값으로 부여하는것은 조금 지양하는 방법입니다. 어떤 값으로 key 속성을 부여해야할 지 조금 더 고민해보세요!

5yoonl commented on src/pages/jinheeKim/Main/Main.js:
```diff
@@ -0,0 +1,226 @@
+                    ))}
+                    <p className="write-time">42초 전</p>
+                    <div className="comment-container">
+                        <TextareaAutosize
+                            type="text"
+                            className="comment-box"
+                            placeholder="댓글달기 .."
+                            value={comment}
+                            onChange={saveComment}
+                            cacheMeasurements
+                        />
```
굳이 외부 라이브러리를 사용하지않더라도 `<textarea></textarea>`태그를 사용하면 multiline input을 활용할 수 있습니다.

5yoonl commented on src/pages/jinheeKim/Main/Main.scss:
```diff
@@ -0,0 +1,265 @@
+@import '../../../style/reset.scss';
+@import '../../../style/variables.scss';
+@font-face {
+	font-family: "Lobster";
+	src: url('../assets/Lobster-Regular.ttf');
+}
+
+$point-color-blue: rgba(47, 170, 249, 1);
+$grey-color-193: rgba(193, 193, 193, 1);
+$grey-color-border: rgba(219, 219, 219, 1);
+$grey-color-bg: rgba(250, 250, 250, 1);
```
앞서 남겨드린 내용의 리뷰와 동일합니다

5yoonl reviewed (commented):
과제진행하느라 수고많으셨습니다 진희님! 리뷰내용이 조금 많다보니 당장 과제를 또 진행하기보다 앞으로 코드를 작성하실 때 지켜야 할 규칙적인 리뷰내용도 있으니 한번 확인해보시고, 코드 수정도 해보신 뒤 계속해서 과제 진행해주시면 되겠습니다!🔥

jininii committed:
Refactor : 멘토 리뷰 반영
- 로그인 페이지 주석 내용 정리
- scss 파일 index.js에 import하여 전역으로 적용
- font 속성 common.scss으로 이동, 변수를 variables.scss로 이동, 전역으로 적용

jininii committed:
Refactor : 멘토 리뷰 반영
- className을 camelCase로 수정

jininii committed:
Refactor : 멘토 리뷰 반영
- comment 관련 변수명 변경
- commentValue 값이 비어있을 때 함수 호출 시 alert 표시 및 댓글 업로드 안되도록 수정
- key 속성 제거
- textarea 외부 라이브러리를 textarea 태그로 변경

jininii committed:
Modify : 상수데이터 활용하여 Aside 하단 내용 변경
Modify : Mock Data로 여러 개의 피드 데이터가 나오도록 수정
Add : 서버 API를 사용하여 회원가입 및 로그인 기능 구현 과제 진행
